.github/workflows/rust.yml

```diff
@@ -15,25 +21,40 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Set up cargo/rust
-        run: |
-          rustup component add rustfmt
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+          override: true
 
       - name: Setup OCaml (because of ocaml-gen)
-        run: sudo apt update && sudo apt install ocaml
+        run: |
+          sudo apt update
+          sudo apt install -y ocaml
 
-      - name: build
+      # `cargo check` compiles the packages without performing the final step of code generation,
+      # which is faster than running `cargo build`.
+      # Use release build to speed up tests
+      - name: build & test
```

Comment:
Can you rename this to `check build` or similar?

Reply:
Sure, or I can create a separate PR for this, as suggested by Joseph

Reply:
@mrmr1993  Here u go https://github.com/o1-labs/proof-systems/pull/332  I will resolve ur comments in the new PR and revert all CI related changes in this one,.